### PR TITLE
Global Settings

### PR DESCRIPTION
**Pull request by gw2princeps**
This implements a global settings menu: 
- Moved language selection in there. 
- moved expert mode selection in there.
- added option to toggle between raid and fractal mode. Raid mode hides AR related options and changes the background.

### CODE REVIEW

**Comment by gw2princeps** [comment]
The way I restore the settings from localStorage is extremely bad. Is there a way to achieve that without redux-persist?

**Comment by marcustyphoon** [comment]
Looking nice! I definitely think it would be neat to have `optimizer.discretize.eu/raids` auto load raids mode. Too bad vite doesn't give us a router - I think reach or react-router just let you say "all of these routes go to the same actual page."

> The way I restore the settings from localStorage is extremely bad. Is there a way to achieve that without redux-persist?

Hmm. You could simplify this API a bit if you wanted to by using a redux key or useState for the current language and just having two useEffect hooks: one with no dependencies that queries localstorage on startup and dispatches whatever is in it, and another with the three redux values as dependencies that saves them all to localstorage. That would automatically fire the hook whenever any of them were changed, so you wouldn't ever need to manually call `saveToLocalstorage`. Other than that, I dunno!

random notes:
 - build page language select
 - should make sure importing a template from before this change defaults to fractal mode
 - I would probably use `raids` internally and just use the fun version for the UI lol

File: src/state/slices/userSettings.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+const defaultState = { expertMode: true, gameMode: 'fractals', language: 'en' };
+// eslint-disable-next-line react-hooks/rules-of-hooks
+const gameModeParam = useQueryParam({ key: PARAMS.GAMEMODE });
+
+// settings grabbed from the local storage. Not the final settings, there are overrides!
+const localStorageSettings = JSON.parse(localStorage.getItem(SETTINGS_STORAGE_KEY));
+export const loadedSettings = {
+  ...defaultState,
+  ...trycatch(() => localStorageSettings, {}), // override default state with potentially saved localStorage variables
```
Comment (marcustyphoon) [inline src/state/slices/userSettings.js:22]: The JSON.parse needs to be in this callback for errors to be caught!

File: src/components/baseComponents/LanguageSelection.jsx [hunk 2]
```diff
@@ -8,34 +8,31 @@ const LANGUAGES = [
   { value: 'de', label: 'Deutsch' },
 ];
 
-const LanguageSelection = () => {
+const LanguageSelection = ({ onChange }) => {
```
Comment (marcustyphoon) [inline src/components/baseComponents/LanguageSelection.jsx:11]: This parameter is unused now that you're using a single useEffect hook to set local storage!

File: src/pages/build/index.jsx [hunk 3]
```diff
@@ -1,24 +1,37 @@
...
-        <LanguageSelection />
-        <Typography variant="h3" sx={{ paddingBottom: 2 }}>
-          <Trans>Shared Build</Trans>
-        </Typography>
+        <Box display="flex" gap={1} alignItems="flex-end" pb={2}>
+          <Typography variant="h3">
+            <Trans>Shared Build</Trans>
+          </Typography>
+          <Box flexGrow={1}>
+            <Chip label={GAME_MODES(t).find((mode) => mode.value === gameMode)?.label} />
```
Comment (marcustyphoon) [inline src/pages/build/index.jsx:28]: Vertical alignment could be tweaked a bit, nbd though

File: src/state/sagas/buildPageSaga.js [hunk 4]
```diff
@@ -112,6 +117,9 @@ function* importStateCharacter({ buildUrl: input, version }) {
 
     const { result } = yield take(decompressChannel);
 
+    // guess the game mode based on agony resistance
```
Comment (marcustyphoon) [inline src/state/sagas/buildPageSaga.js:120]: (discussed in discord)

File: src/state/sagas/buildPageSaga.js [hunk 5]
```diff
@@ -6,15 +6,17 @@ import {
...
+import { changeGameMode } from '../slices/userSettings';
 import SagaTypes from './sagaTypes';
 
 // channels solve the problem "how to get value out of callback"
 // https://stackoverflow.com/questions/43031832/how-to-yield-put-in-redux-saga-within-a-callback
 const compressChannel = channel();
 function* exportStateCharacter({ newPage, copyToClipboard }) {
   const { optimizer } = yield select();
-  const { buildPage, control } = optimizer;
+  const { buildPage, control, userSettings } = optimizer;
```
Comment (marcustyphoon) [inline src/state/sagas/buildPageSaga.js:19]: in spots like this you can do `const state = yield select()` and import selectors and call them on `state` if you want to decouple the code from the slice shape; I don't think it really matters but it's a noteworthy option for when the slice is really complicated

**Comment by marcustyphoon** [comment]
Re: hoverable info boxes, cool dual functionality but I'm not sure if it would get used much compared to clicking? I could go either way on that one.

File: src/state/slices/userSettings.js (hunk 1 above)
Comment (gw2princeps) [reply src/state/slices/userSettings.js:22]: oh my god - it used to be there, but I tried some things and forgot to put it back

**Comment by gw2princeps** [comment]
The info boxes are also clickable

**Review by marcustyphoon — APPROVED** [review]

**Pull request merged**